src/main/java/com/minecolonies/api/util/ItemStackUtils.java

```diff
@@ -1098,5 +1105,55 @@ public static Set<ItemStack> allItemsPlusInventory(@NotNull final Player player)
 
         return allItems.stream().map(ItemStorage::getItemStack).collect(Collectors.toSet());
     }
+
+    /**
+     * Consume food helper.
+     *
+     * @param foodStack   the itemstack of food.
+     * @param citizen     the citizen entity.
+     * @param citizenData the citizen data.
+     * @param inventory optional inventory to insert stack into if not citizen.
+     */
+    public static void consumeFood(final ItemStack foodStack, final AbstractEntityCitizen citizen, final ICitizenData citizenData, final Inventory inventory)
```

Comment:
just need either citizen data or citizen entity as input

Reply:
I know we can technically get it from either, but we need an extra query step which we already done at the places where we call this.

Reply:
ye you make it one param to avoid potential futures issues where someone might use it with a desync of worker/data